index.js

```diff
@@ -25,13 +27,57 @@ elixir.extend('wiredep', function(config, opts) {
             js: '<script src="{{filePath}}"></script>',
             css: '<link rel="stylesheet" href="{{filePath}}" />'
           }
-        }
+        },
+        less: {
+          block: /(([ \t]*)\/\/\s*bower:*(\S*))(\n|\r|.)*?(\/\/\s*endbower)/gi,
+          detect: {
+              css: /@import\s['"](.+css)['"]/gi,
+              less: /@import\s['"](.+less)['"]/gi
+          },
+          replace: {
+              css: '@import "{{filePath}}";',
+              less: '@import "{{filePath}}";'
+          }
+        },
+        sass: {
+          block: /(([ \t]*)\/\/\s*bower:*(\S*))(\n|\r|.)*?(\/\/\s*endbower)/gi,
+          detect: {
+              css: /@import\s(.+css)/gi,
+              sass: /@import\s(.+sass)/gi,
+              scss: /@import\s(.+scss)/gi
+          },
+          replace: {
+              css: '@import {{filePath}}',
+              sass: '@import {{filePath}}',
+              scss: '@import {{filePath}}'
+          }
+        },
+        scss: {
+          block: /(([ \t]*)\/\/\s*bower:*(\S*))(\n|\r|.)*?(\/\/\s*endbower)/gi,
+          detect: {
+              css: /@import\s['"](.+css)['"]/gi,
+              sass: /@import\s['"](.+sass)['"]/gi,
+              scss: /@import\s['"](.+scss)['"]/gi
+          },
+          replace: {
+              css: '@import "{{filePath}}";',
+              sass: '@import "{{filePath}}";',
+              scss: '@import "{{filePath}}";'
+          }
+        },
       };
 
+      var less = path.join(econ.get('assets.less.folder'), '**/*.less');
+      var sass = path.join(econ.get('assets.sass.folder'), '**/*.sass');
+      var scss = path.join(econ.get('assets.sass.folder'), '**/*.scss');
+
       new task('wiredep', function() {
         var src = path.join(config.baseDir, !!config.src ? config.src : config.searchLevel);
-
-        return gulp.src(src).pipe(wiredep(opts)).pipe(gulp.dest(config.baseDir))
+        return merge( gulp.src(less).pipe(wiredep(opts)).pipe(gulp.dest(econ.get('assets.css.less.folder'))),
+                      gulp.src(sass).pipe(wiredep(opts)).pipe(gulp.dest(econ.get('assets.css.sass.folder'))),
+                      gulp.src(scss).pipe(wiredep(opts)).pipe(gulp.dest(econ.get('assets.css.sass.folder'))),
+                      gulp.src(src).pipe(wiredep(opts)).pipe(gulp.dest(config.baseDir))
+                    );
       })
```

Comment:
I didn't like this solution because you are trying to mix 3 different files when most of the types you just want to wiredep Sass or Scss or Less files